lib/app/analytics.coffee

```diff
@@ -0,0 +1,23 @@
+opts = require '../options'
+Analytics = require 'analytics-node'
+
+@setCampaign = (req, res, next) ->
+  return next() unless opts.SEGMENT_WRITE_KEY
+  req.session.modalId = req.params.modal_id
+  req.session.acquisitionInitiative = req.params.acquisition_initiative
```

Comment:
Yeah, for Facebook and Twitter signup we'll end up redirecting to outside Artsy losing our state. Alternatively, I think we can pass params to FB/Twitter that we could catch again when coming back to Artsy. It might be worth looking into that refactor, but this seemed like the most obvious way to do it.

This is primarily used to track which "campaign" the user signed up through. For context on what I'm talking about wrt to "campaigns" see this PR https://github.com/artsy/force/pull/393